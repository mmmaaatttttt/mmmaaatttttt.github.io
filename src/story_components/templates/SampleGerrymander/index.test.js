import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { SampleGerrymander } from ".";

describe("smoke and snapshot tests", () => {
  it("renders successfully", () => {
    shallow(<SampleGerrymander />);
  });

  it("renders first demonstration successfully", () => {
    const wrapper = shallow(<SampleGerrymander />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe("state changes", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SampleGerrymander />);
  });

  it("detects the formation of new districts", () => {
    const { segments } = wrapper.state();
    const row = 1;
    for (let i = 0; i < segments[row].length; i++) {
      expect(wrapper.state().districts).toHaveLength(1);
      expect(wrapper.state().districts[0]).toHaveLength(54);
      wrapper.instance().handleSegmentUpdate(1, i, true);
    }
    expect(wrapper.state().districts).toHaveLength(2);
    expect(wrapper.state().districts[0]).toHaveLength(9);
    expect(wrapper.state().districts[1]).toHaveLength(45);
  });

  it("detects when there are too many districts", () => {
    const { segments } = wrapper.state();
    // make horizontal row districts
    const newSegments = segments.map((row, i) =>
      i % 2 === 0 ? row : row.map(bool => !bool)
    );
    wrapper.setState({ segments: newSegments });
    wrapper.instance().__countRegions();
    expect(wrapper.find("Icon")).toHaveLength(6);

    newSegments[0][1] = true;
    wrapper.setState({ segments: newSegments });
    wrapper.instance().__countRegions();
    expect(wrapper.find("Icon")).toHaveLength(0);
    expect(wrapper.text()).toContain("Too many districts!");
  });

  it("detects when a district is the right size", () => {
    const { segments } = wrapper.state();
    const newSegments = segments.map((row, i) =>
      i !== 1 ? row : row.map(bool => !bool)
    );

    expect(wrapper.find("Icon[name='check-circle']")).toHaveLength(0);
    wrapper.setState({ segments: newSegments });
    wrapper.instance().__countRegions();
    expect(wrapper.find("Icon[name='check-circle']")).toHaveLength(1);
  });

  it("detects when a district is the wrong size", () => {
    const { segments } = wrapper.state();

    const newSegments = segments.map(row => [...row]);
    newSegments[0][0] = true;
    newSegments[1][0] = true;

    expect(wrapper.find("Icon[name='check-circle']")).toHaveLength(0);
    expect(wrapper.state().districts).toHaveLength(1);
    wrapper.setState({ segments: newSegments });
    wrapper.instance().__countRegions();
    expect(wrapper.state().districts).toHaveLength(2);
    expect(wrapper.find("Icon[name='check-circle']")).toHaveLength(0);
  });
});
