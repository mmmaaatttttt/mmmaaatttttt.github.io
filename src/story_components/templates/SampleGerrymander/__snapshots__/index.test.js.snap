// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests renders first demonstration successfully 1`] = `
<styled.div
  break="small"
  sizes={
    Array [
      3,
      2,
    ]
  }
>
  <HeatChart
    accessor={[Function]}
    axes={false}
    colorDomain={
      Array [
        0,
        1,
      ]
    }
    colorRange={
      Array [
        "#2227FF",
        "#FF3C23",
      ]
    }
    data={
      Array [
        Array [
          0,
          1,
          0,
          1,
          0,
          1,
        ],
        Array [
          0,
          1,
          0,
          1,
          0,
          1,
        ],
        Array [
          0,
          1,
          0,
          1,
          0,
          1,
        ],
        Array [
          0,
          1,
          0,
          1,
          0,
          1,
        ],
        Array [
          0,
          1,
          0,
          1,
          0,
          1,
        ],
        Array [
          0,
          1,
          0,
          1,
          0,
          1,
        ],
        Array [
          0,
          1,
          0,
          1,
          0,
          1,
        ],
        Array [
          0,
          1,
          0,
          1,
          0,
          1,
        ],
        Array [
          0,
          1,
          0,
          1,
          0,
          1,
        ],
      ]
    }
    getTooltipBody={[Function]}
    tooltip={false}
  >
    <InteractiveGrid
      colCount={9}
      handleSegmentUpdate={[Function]}
      height={600}
      paddingX={45}
      paddingY={45}
      rowCount={6}
      segments={
        Array [
          Array [
            false,
            false,
            false,
            false,
            false,
            false,
            false,
            false,
          ],
          Array [
            false,
            false,
            false,
            false,
            false,
            false,
            false,
            false,
            false,
          ],
          Array [
            false,
            false,
            false,
            false,
            false,
            false,
            false,
            false,
          ],
          Array [
            false,
            false,
            false,
            false,
            false,
            false,
            false,
            false,
            false,
          ],
          Array [
            false,
            false,
            false,
            false,
            false,
            false,
            false,
            false,
          ],
          Array [
            false,
            false,
            false,
            false,
            false,
            false,
            false,
            false,
            false,
          ],
          Array [
            false,
            false,
            false,
            false,
            false,
            false,
            false,
            false,
          ],
          Array [
            false,
            false,
            false,
            false,
            false,
            false,
            false,
            false,
            false,
          ],
          Array [
            false,
            false,
            false,
            false,
            false,
            false,
            false,
            false,
          ],
          Array [
            false,
            false,
            false,
            false,
            false,
            false,
            false,
            false,
            false,
          ],
          Array [
            false,
            false,
            false,
            false,
            false,
            false,
            false,
            false,
          ],
        ]
      }
      strokeWidth={6}
      width={600}
    />
  </HeatChart>
  <styled.div>
    <div
      key="0"
    >
      <h4
        style={
          Object {
            "color": "#000000",
          }
        }
      >
        D
        1
        : 
        54
         
        (27 blue, 27 red)
      </h4>
      <Icon
        color="#FF3C23"
        disabled={false}
        name="times-circle"
        opacity={1}
        size={2}
        type="solid"
      />
    </div>
    <div
      key="1"
    >
      <h4
        style={
          Object {
            "color": "#000000",
          }
        }
      >
        D
        2
        : 
        --
         
      </h4>
      <Icon
        color="#FF3C23"
        disabled={false}
        name="times-circle"
        opacity={1}
        size={2}
        type="solid"
      />
    </div>
    <div
      key="2"
    >
      <h4
        style={
          Object {
            "color": "#000000",
          }
        }
      >
        D
        3
        : 
        --
         
      </h4>
      <Icon
        color="#FF3C23"
        disabled={false}
        name="times-circle"
        opacity={1}
        size={2}
        type="solid"
      />
    </div>
    <div
      key="3"
    >
      <h4
        style={
          Object {
            "color": "#000000",
          }
        }
      >
        D
        4
        : 
        --
         
      </h4>
      <Icon
        color="#FF3C23"
        disabled={false}
        name="times-circle"
        opacity={1}
        size={2}
        type="solid"
      />
    </div>
    <div
      key="4"
    >
      <h4
        style={
          Object {
            "color": "#000000",
          }
        }
      >
        D
        5
        : 
        --
         
      </h4>
      <Icon
        color="#FF3C23"
        disabled={false}
        name="times-circle"
        opacity={1}
        size={2}
        type="solid"
      />
    </div>
    <div
      key="5"
    >
      <h4
        style={
          Object {
            "color": "#000000",
          }
        }
      >
        D
        6
        : 
        --
         
      </h4>
      <Icon
        color="#FF3C23"
        disabled={false}
        name="times-circle"
        opacity={1}
        size={2}
        type="solid"
      />
    </div>
    <div>
      <Button
        color="#52A081"
        disabled={true}
        large={false}
        onClick={[Function]}
      >
        Saved
      </Button>
      <Button
        color="#FF3C23"
        disabled={false}
        large={false}
        onClick={[Function]}
      >
        Reset
      </Button>
    </div>
  </styled.div>
</styled.div>
`;
